### PR TITLE
MMT-2416 Adding Collection-Tool Associations

### PR DESCRIPTION
**Pull request by ryanmiller-1**
Added manage associations link to published tool page.
Added correct underlining when looking at tool association pages.
Updated collection associations controller and change provider modal
Added associated routes
Updated cmr_client to share code for service/tool associations
Copied tests from service associations and updated for tools

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -227,11 +227,11 @@ en:
         destroy:
```
Comment (charleshuang80) [inline config/locales/en.yml:227]: if these actions are not possible anymore, should we remove the messages here?

**Review by charleshuang80 — APPROVED** [review]

File: lib/cmr/cmr_client.rb [hunk 2]
```diff
@@ -99,10 +99,18 @@ def test_query(query, token, headers = {})
...
 
     def add_collection_assocations_to_service(concept_id, collection_ids, token)
+      add_collection_assocations(concept_id, collection_ids, token, 'services')
+    end
+
+    def add_collection_assocations_to_tool(concept_id, collection_ids, token)
+      add_collection_assocations(concept_id, collection_ids, token, 'tools')
+    end
+
+    def add_collection_assocations(concept_id, collection_ids, token, concept_type)
```
Comment (ChristianTrummer99) [inline lib/cmr/cmr_client.rb:109]: You might consider adding something like `concept_type = (concept_id[0] == 'T' ? 'tools' : 'services')` instead of using different methods. The way I see it, the `add_collection_associations` method is only used for non-collection metadata so maintenance would be low if we just skimmed the first character of the concept-id. Your methods are a bit more readable though, so either way works

File: lib/cmr/cmr_client.rb [hunk 3]
```diff
@@ -113,10 +121,18 @@ def add_collection_assocations_to_service(concept_id, collection_ids, token)
...
 
     def delete_collection_assocations_to_service(concept_id, collection_ids, token)
+      delete_collection_associations(concept_id, collection_ids, token, 'services')
+    end
+
+    def delete_collection_assocations_to_tool(concept_id, collection_ids, token)
+      delete_collection_associations(concept_id, collection_ids, token, 'tools')
+    end
+
+    def delete_collection_associations(concept_id, collection_ids, token, concept_type)
```
Comment (ChristianTrummer99) [inline lib/cmr/cmr_client.rb:131]: same as above

File: spec/features/services/delete_collection_assocation_spec.rb [hunk 4]
```diff
@@ -59,7 +57,7 @@
             click_link 'refresh the page'
           end
 
-          it 'reloades the page and dislay the correct associations' do
+          it 'reloads the page and dislay the correct associations' do
```
Comment (ChristianTrummer99) [inline spec/features/services/delete_collection_assocation_spec.rb:60]: 'display' is also misspelled

File: spec/features/tools/create_collection_association_spec.rb [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+      before do
+        within '#collection-search' do
+          select 'Entry Title', from: 'Search Field'
+          find(:css, "input[id$='query_text']").set('MODIS-I*')
+
+          click_button 'Submit'
+        end
+      end
+
+      it 'display the correct search results' do
```
Comment (ChristianTrummer99) [inline spec/features/tools/create_collection_association_spec.rb:93]: being picky: grammar

**Review by ChristianTrummer99 — COMMENTED** [review]
- Getting an error banner when I click ‘Manage Associated Collections’ on published tool show page
- Getting actual failure when I click ‘Add Associated Collection’ due to "undefined method 'any?' for nil:NilClass"
- I suggest Rails logging for the `get_all_associated_collections` method for app/controllers/collection_associations_controller.rb:149 because I can't tell if this is due to the conditions not being met or the `association_response` failing
- app/controllers/collection_associations_controller.rb:155 is causing issues because `partial_associated_collections` is `nil`
- The above issues could be due to something ill-configured on my machine given the content of the error banner
- spec/features/tools/create_collection_association_spec.rb is failing locally for me
In summary, I think these issues might be something wrong with my machine, but the handling of `partial_associated_collections` being `nil` should probably be friendlier

File: lib/cmr/cmr_client.rb (hunk 2 above)
Comment (ryanmiller-1) [reply lib/cmr/cmr_client.rb:109]: Resolving this by changing the call in the controller.

**Comment by ryanmiller-1** [comment]
I believe that all of those problems are resolved with an uberjar update.  An old uberjar won't recognize tool_concept_id as a parameter for collection searching and will not have the end points to make tool associations.  Regardless, I have made failure for that collection search more friendly.

**Review by ChristianTrummer99 — APPROVED** [review]
